tests/orchestrator.rs

```diff
@@ -63,7 +69,10 @@ where
     // test delete
     orchestrator.delete_blocking(&pod_run)?;
     assert!(
-        orchestrator.list_blocking()?.is_empty(),
+        !orchestrator
+            .list_blocking()?
+            .iter()
+            .any(|container| container.pod_job == pod_run.pod_job),
```

Comment:
Perhaps we should use filter by pod_run since that would be unique than pod_job. We did discuss this a while back and I believe one of the standing PR does resolve this, so up to you if we should merge it as is.